<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <title>Transcriptome Analysis in R</title>
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    span.underline{text-decoration: underline;}
    div.column{display: inline-block; vertical-align: top; width: 50%;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #aaaaaa;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
    div.sourceCode
      {   }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span.al { color: #ff0000; font-weight: bold; } /* Alert */
    code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
    code span.at { color: #7d9029; } /* Attribute */
    code span.bn { color: #40a070; } /* BaseN */
    code span.bu { } /* BuiltIn */
    code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
    code span.ch { color: #4070a0; } /* Char */
    code span.cn { color: #880000; } /* Constant */
    code span.co { color: #60a0b0; font-style: italic; } /* Comment */
    code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
    code span.do { color: #ba2121; font-style: italic; } /* Documentation */
    code span.dt { color: #902000; } /* DataType */
    code span.dv { color: #40a070; } /* DecVal */
    code span.er { color: #ff0000; font-weight: bold; } /* Error */
    code span.ex { } /* Extension */
    code span.fl { color: #40a070; } /* Float */
    code span.fu { color: #06287e; } /* Function */
    code span.im { } /* Import */
    code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
    code span.kw { color: #007020; font-weight: bold; } /* Keyword */
    code span.op { color: #666666; } /* Operator */
    code span.ot { color: #007020; } /* Other */
    code span.pp { color: #bc7a00; } /* Preprocessor */
    code span.sc { color: #4070a0; } /* SpecialChar */
    code span.ss { color: #bb6688; } /* SpecialString */
    code span.st { color: #4070a0; } /* String */
    code span.va { color: #19177c; } /* Variable */
    code span.vs { color: #4070a0; } /* VerbatimString */
    code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
    .display.math{display: block; text-align: center; margin: 0.5rem auto;}
  </style>
  <link rel="stylesheet" href="avenir-white.css" />
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<header id="title-block-header">
<h1 class="title">Transcriptome Analysis in R</h1>
</header>
<p>This week we will work with a RNA sequencing data set to analyze patterns of differential expression of genes from human X chromosome. This data sets is described in this publication: <a href="data/Pertea_rnaseq_2016.pdf">Pertea et al (2016)</a>.</p>
<p>The data set itself can be downloaded here (<strong>but you do not need to</strong>): <a href="ftp://ftp.ccb.jhu.edu/pub/RNAseq_protocol/chrX_data.tar.gz">chrX_data.tar.gz</a></p>
<p><br><br></p>
<h2 id="table-of-contents">Table of Contents</h2>
<ul>
<li><p><a href="#typical-rnaseq-analysis-pipeline">1. Typical RNAseq Analysis Pipeline</a></p></li>
<li><p><a href="#data-access">2. Data Access</a></p></li>
<li><p><a href="#data-cleanup">3. Data Cleanup</a></p></li>
</ul>
<p><br><br><br><br><br><br><br><br> <br><br><br><br><br><br><br><br> <br><br><br><br><br><br><br><br> <br><br><br><br><br><br><br><br> <br><br><br><br><br><br><br><br></p>
<h2 id="typical-rnaseq-analysis-pipeline">1. Typical RNAseq Analysis Pipeline</h2>
<p>Transcriptomic data from short-read sequencing technologies (e.g. Illumina HiSeq) is analyzed in the following manner:</p>
<ol type="1">
<li><p>Obtain sequencing reads from service provider</p></li>
<li><p>Perform quality control to remove low quality sequencing reads</p></li>
<li><p>Obtain sequence data for a reference genome</p></li>
<li><p>Index the reference genome (so it can be efficiently searched for homologies)</p></li>
<li><p>Align post-QC sequencing reads to the indexed reference genome</p></li>
<li><p>Identify transcripts/genes based on their homologies and locations in the reference genome.</p></li>
<li><p>Calculate the transcript abundances in each sample under study and prepare a table of gene and transcript counts.</p></li>
<li><p>Use regression analysis to determine which genes/transcripts are up and down regulated.</p></li>
<li><p>Form hypotheses and test them using the differential expression results.</p></li>
</ol>
<p>Because this is a R tutorial, we will not get into analysis steps 1 through 7. Please consult the publication linked above for the details of those analyses.</p>
<p>We will begin with step #8 above.</p>
<h2 id="data-access">2. Data Access</h2>
<p>In any RNA sequencing experiment, there are two main types of data: genomic and phenotypic. The former comes from sequencing and counting transcripts/genes, and the latter is trait data that will be used to make meaningful inferences from the results.</p>
<h3 id="phenotypic-data">2.1 Phenotypic Data</h3>
<center>
<img src="data/phenotypes.png" width=400>
</center>
<p>The data above can’t be copy pasted. It is an image. We will need to create this data frame in R manually.</p>
<p><br><br></p>
<ol type="1">
<li>First create 3 vectors</li>
</ol>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true"></a>sampid &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;ERR188245&quot;</span>, <span class="st">&quot;ERR188428&quot;</span>, <span class="st">&quot;ERR188337&quot;</span>, <span class="st">&quot;ERR188401&quot;</span>, <span class="st">&quot;ERR188257&quot;</span>,</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true"></a><span class="st">&quot;ERR188383&quot;</span>, <span class="st">&quot;ERR204916&quot;</span>, <span class="st">&quot;ERR188234&quot;</span>, <span class="st">&quot;ERR188273&quot;</span>, <span class="st">&quot;ERR188454&quot;</span>, <span class="st">&quot;ERR188104&quot;</span>,</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true"></a><span class="st">&quot;ERR188044&quot;</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true"></a></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true"></a>sex &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="kw">replicate</span>(<span class="dv">3</span>, <span class="st">&quot;Female&quot;</span>), <span class="kw">replicate</span>(<span class="dv">3</span>, <span class="st">&quot;Male&quot;</span>), <span class="kw">replicate</span>(<span class="dv">3</span>, <span class="st">&quot;Female&quot;</span>),</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true"></a><span class="kw">replicate</span>(<span class="dv">3</span>, <span class="st">&quot;Male&quot;</span>))</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true"></a></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true"></a>pop &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="kw">replicate</span>(<span class="dv">6</span>, <span class="st">&quot;GBR&quot;</span>), <span class="kw">replicate</span>(<span class="dv">6</span>, <span class="st">&quot;YRI&quot;</span>))</span></code></pre></div>
<p><br><br></p>
<ol start="2" type="1">
<li>Check to make sure all vectors are of same length</li>
</ol>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true"></a><span class="kw">length</span>(sampid)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true"></a><span class="kw">length</span>(sex)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true"></a><span class="kw">length</span>(pop)</span></code></pre></div>
<p><br><br></p>
<ol start="3" type="1">
<li>Creat a data frame by combining the three vectors as columns:</li>
</ol>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true"></a>pheno &lt;-<span class="st"> </span><span class="kw">data.frame</span>(sampid, sex, pop)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true"></a>pheno</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true"></a></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true"></a>      sampid    sex pop</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true"></a><span class="dv">1</span>  ERR188245 Female GBR</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true"></a><span class="dv">2</span>  ERR188428 Female GBR</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true"></a><span class="dv">3</span>  ERR188337 Female GBR</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true"></a><span class="dv">4</span>  ERR188401   Male GBR</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true"></a><span class="dv">5</span>  ERR188257   Male GBR</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true"></a><span class="dv">6</span>  ERR188383   Male GBR</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true"></a><span class="dv">7</span>  ERR204916 Female YRI</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true"></a><span class="dv">8</span>  ERR188234 Female YRI</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true"></a><span class="dv">9</span>  ERR188273 Female YRI</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true"></a><span class="dv">10</span> ERR188454   Male YRI</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true"></a><span class="dv">11</span> ERR188104   Male YRI</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true"></a><span class="dv">12</span> ERR188044   Male YRI</span></code></pre></div>
<p><br><br></p>
<h3 id="gene-count-data">2.2 Gene Count Data</h3>
<ol type="1">
<li>The gene count data for each sample can be downloaded here: <a href="data/gene_count_matrix.csv">gene_count_matrix.csv</a>. Once downloaded, import the data into R:</li>
</ol>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true"></a>gc &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;gene_count_matrix.csv&quot;</span>, <span class="dt">header=</span><span class="ot">TRUE</span>)</span></code></pre></div>
<ol start="2" type="1">
<li>What are the dimensions of this data frame?</li>
</ol>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true"></a><span class="kw">dim</span>(gc)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true"></a>[<span class="dv">1</span>] <span class="dv">2082</span>   <span class="dv">13</span></span></code></pre></div>
<ol start="3" type="1">
<li>Check sample names to make sure they match the <code>sampid</code> from the <code>pheno</code> data you created above.</li>
</ol>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true"></a><span class="kw">names</span>(gc)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true"></a> [<span class="dv">1</span>] <span class="st">&quot;gene_id&quot;</span>   <span class="st">&quot;ERR188044&quot;</span> <span class="st">&quot;ERR188104&quot;</span> <span class="st">&quot;ERR188234&quot;</span> <span class="st">&quot;ERR188245&quot;</span> <span class="st">&quot;ERR188257&quot;</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true"></a> [<span class="dv">7</span>] <span class="st">&quot;ERR188273&quot;</span> <span class="st">&quot;ERR188337&quot;</span> <span class="st">&quot;ERR188383&quot;</span> <span class="st">&quot;ERR188401&quot;</span> <span class="st">&quot;ERR188428&quot;</span> <span class="st">&quot;ERR188454&quot;</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true"></a>[<span class="dv">13</span>] <span class="st">&quot;ERR204916&quot;</span></span></code></pre></div>
<ul>
<li><p>Did you notice that the sample names are not in the same order in <code>gc</code> compared to <code>pheno</code>. This is problematic. The analytical software we are going to use requires that both files should have the same order for sample names. How do we fix this?</p></li>
<li><p>We can either change the order in one or the other file. In the pheno file, the order is appropriate for the analysis, which clearly shows replicates for each experimental condition. So we will keep that file as is and change <code>gc</code> instead.</p></li>
</ul>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true"></a>gc2 &lt;-<span class="st"> </span>gc[,<span class="kw">c</span>(<span class="st">&quot;gene_id&quot;</span>, sampid)]</span></code></pre></div>
<ul>
<li>Note what did above. We already have the correct order of samples in the vector <code>sampid</code> we created earlier. So we simply substituted it and simply pre-pended <code>gene_id</code> to that list. Check to make sure that the new data frame shows the data correctly:</li>
</ul>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true"></a><span class="kw">head</span>(gc)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true"></a>               gene_id ERR188044 ERR188104 ERR188234 ERR188245 ERR188257 ERR188273 ERR188337 ERR188383 ERR188401 ERR188428 ERR188454 ERR204916</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true"></a><span class="dv">1</span>  NM_<span class="dv">001177465</span><span class="op">|</span>MAMLD1         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true"></a><span class="dv">2</span>      NR_<span class="dv">029427</span><span class="op">|</span>WDR13         <span class="dv">0</span>         <span class="dv">0</span>        <span class="dv">54</span>       <span class="dv">211</span>       <span class="dv">204</span>        <span class="dv">47</span>       <span class="dv">177</span>       <span class="dv">165</span>       <span class="dv">177</span>       <span class="dv">112</span>       <span class="dv">169</span>       <span class="dv">191</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true"></a><span class="dv">3</span>   NM_<span class="dv">001242640</span><span class="op">|</span>BRCC3       <span class="dv">401</span>       <span class="dv">495</span>       <span class="dv">574</span>       <span class="dv">520</span>       <span class="dv">614</span>        <span class="dv">83</span>       <span class="dv">737</span>       <span class="dv">304</span>       <span class="dv">737</span>       <span class="dv">313</span>       <span class="dv">358</span>       <span class="dv">312</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true"></a><span class="dv">4</span>   NM_<span class="dv">001303516</span><span class="op">|</span>PDZD4         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true"></a><span class="dv">5</span> NM_<span class="dv">001170757</span><span class="op">|</span>FAM122B       <span class="dv">865</span>       <span class="dv">470</span>       <span class="dv">462</span>         <span class="dv">0</span>       <span class="dv">271</span>        <span class="dv">94</span>       <span class="dv">342</span>       <span class="dv">295</span>       <span class="dv">342</span>       <span class="dv">306</span>       <span class="dv">403</span>       <span class="dv">127</span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true"></a><span class="dv">6</span>    NM_<span class="dv">001185081</span><span class="op">|</span>FMR1         <span class="dv">0</span>        <span class="dv">95</span>         <span class="dv">0</span>         <span class="dv">0</span>       <span class="dv">346</span>         <span class="dv">0</span>        <span class="dv">82</span>       <span class="dv">294</span>        <span class="dv">82</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true"></a></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true"></a></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true"></a><span class="kw">head</span>(gc2)</span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true"></a></span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true"></a>               gene_id ERR188245 ERR188428 ERR188337 ERR188401 ERR188257 ERR188383 ERR204916 ERR188234 ERR188273 ERR188454 ERR188104 ERR188044</span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true"></a><span class="dv">1</span>  NM_<span class="dv">001177465</span><span class="op">|</span>MAMLD1         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span></span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true"></a><span class="dv">2</span>      NR_<span class="dv">029427</span><span class="op">|</span>WDR13       <span class="dv">211</span>       <span class="dv">112</span>       <span class="dv">177</span>       <span class="dv">177</span>       <span class="dv">204</span>       <span class="dv">165</span>       <span class="dv">191</span>        <span class="dv">54</span>        <span class="dv">47</span>       <span class="dv">169</span>         <span class="dv">0</span>         <span class="dv">0</span></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true"></a><span class="dv">3</span>   NM_<span class="dv">001242640</span><span class="op">|</span>BRCC3       <span class="dv">520</span>       <span class="dv">313</span>       <span class="dv">737</span>       <span class="dv">737</span>       <span class="dv">614</span>       <span class="dv">304</span>       <span class="dv">312</span>       <span class="dv">574</span>        <span class="dv">83</span>       <span class="dv">358</span>       <span class="dv">495</span>       <span class="dv">401</span></span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true"></a><span class="dv">4</span>   NM_<span class="dv">001303516</span><span class="op">|</span>PDZD4         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span></span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true"></a><span class="dv">5</span> NM_<span class="dv">001170757</span><span class="op">|</span>FAM122B         <span class="dv">0</span>       <span class="dv">306</span>       <span class="dv">342</span>       <span class="dv">342</span>       <span class="dv">271</span>       <span class="dv">295</span>       <span class="dv">127</span>       <span class="dv">462</span>        <span class="dv">94</span>       <span class="dv">403</span>       <span class="dv">470</span>       <span class="dv">865</span></span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true"></a><span class="dv">6</span>    NM_<span class="dv">001185081</span><span class="op">|</span>FMR1         <span class="dv">0</span>         <span class="dv">0</span>        <span class="dv">82</span>        <span class="dv">82</span>       <span class="dv">346</span>       <span class="dv">294</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>         <span class="dv">0</span>        <span class="dv">95</span>         <span class="dv">0</span></span></code></pre></div>
<ul>
<li>At this point our data sets should be ready for further steps.</li>
</ul>
<p><br><br></p>
<h2 id="data-cleanup">3. Data Cleanup</h2>
<ul>
<li>One thing you may have noticed is that we seem to have many genes where there is no data for any of the 12 samples. These are the genes which read <code>0</code> in each of the 12 columns. There is essentially no data here, so we need to get rid of those rows.</li>
</ul>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true"></a>gc3 &lt;-<span class="st"> </span>gc2[<span class="kw">apply</span>(gc2[<span class="kw">c</span>(<span class="dv">2</span><span class="op">:</span><span class="dv">13</span>)], <span class="dv">1</span>, <span class="cf">function</span>(x) <span class="op">!</span><span class="kw">any</span>(x <span class="op">==</span><span class="st"> </span><span class="dv">0</span>)),]</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true"></a><span class="kw">dim</span>(gc3)</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true"></a></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true"></a>[<span class="dv">1</span>] <span class="dv">656</span>  <span class="dv">13</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true"></a></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true"></a><span class="kw">head</span>(gc3)</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true"></a></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true"></a>              gene_id ERR188245 ERR188428 ERR188337 ERR188401 ERR188257 ERR188383 ERR204916 ERR188234 ERR188273 ERR188454 ERR188104 ERR188044</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true"></a><span class="dv">3</span>  NM_<span class="dv">001242640</span><span class="op">|</span>BRCC3       <span class="dv">520</span>       <span class="dv">313</span>       <span class="dv">737</span>       <span class="dv">737</span>       <span class="dv">614</span>       <span class="dv">304</span>       <span class="dv">312</span>       <span class="dv">574</span>        <span class="dv">83</span>       <span class="dv">358</span>       <span class="dv">495</span>       <span class="dv">401</span></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true"></a><span class="dv">7</span>     NM_<span class="dv">001356</span><span class="op">|</span>DDX3X       <span class="dv">203</span>       <span class="dv">336</span>       <span class="dv">312</span>       <span class="dv">312</span>       <span class="dv">239</span>        <span class="dv">37</span>       <span class="dv">129</span>       <span class="dv">432</span>        <span class="dv">44</span>        <span class="dv">32</span>       <span class="dv">879</span>       <span class="dv">102</span></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true"></a><span class="dv">13</span>     NM_<span class="dv">000489</span><span class="op">|</span>ATRX      <span class="dv">1260</span>      <span class="dv">3278</span>      <span class="dv">3919</span>      <span class="dv">3919</span>      <span class="dv">4098</span>      <span class="dv">3632</span>      <span class="dv">4259</span>      <span class="dv">1946</span>      <span class="dv">1692</span>      <span class="dv">2501</span>      <span class="dv">5921</span>      <span class="dv">4909</span></span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true"></a><span class="dv">22</span>      NM_<span class="dv">000116</span><span class="op">|</span>TAZ       <span class="dv">207</span>       <span class="dv">282</span>       <span class="dv">562</span>       <span class="dv">562</span>       <span class="dv">286</span>       <span class="dv">389</span>       <span class="dv">468</span>       <span class="dv">598</span>       <span class="dv">130</span>       <span class="dv">263</span>       <span class="dv">430</span>       <span class="dv">432</span></span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true"></a><span class="dv">23</span>    NM_<span class="dv">016024</span><span class="op">|</span>RBMX2      <span class="dv">1072</span>      <span class="dv">1164</span>      <span class="dv">1849</span>      <span class="dv">1849</span>      <span class="dv">1356</span>      <span class="dv">1752</span>      <span class="dv">1430</span>      <span class="dv">1453</span>       <span class="dv">765</span>      <span class="dv">1292</span>      <span class="dv">1843</span>      <span class="dv">1626</span></span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true"></a><span class="dv">28</span>   NR_<span class="dv">029635</span><span class="op">|</span>MIR221         <span class="dv">1</span>         <span class="dv">2</span>         <span class="dv">8</span>         <span class="dv">8</span>         <span class="dv">5</span>         <span class="dv">2</span>         <span class="dv">3</span>         <span class="dv">3</span>         <span class="dv">2</span>         <span class="dv">2</span>        <span class="dv">10</span>         <span class="dv">6</span></span></code></pre></div>
<ul>
<li>What we did above is to remove rows containing only 0s for columns 2 through 14 in the data set. But instead of saving it in the same data frame, we sent output to a new frame called <code>gc3</code>. We now have 656 rows of data, which means 1426 rows had no data, so we discarded them.</li>
</ul>
<p><br><br><br><br> <br><br><br><br> <br><br><br><br> <br><br><br><br></p>
</body>
</html>
